### PR TITLE
fix: use FeatureFlags in @superset-ui/core

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Feature flag typing [has been added to `@superset-ui/core`](https://github.com/apache-superset/superset-ui/pull/982), since there is now two copies of the same typing, sometimes it leads to TS errors when `npm link` is used or when in Jest.

This change means all future feature flags used in the front-end have to be added to `@superset-ui` and go through the npm publish process. This is inconvenient but probably necessary to ensure correctness and consistency. Considering we are merging `superset-ui` to this repo soon, this should not be that much of a problem.

cc @graceguo-supercat @villebro 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TEST PLAN

1. `npm link ../superset-ui/packages/superset-ui-core`
2. `npm run type`
3. You will see this error in `master`: 

```
node_modules/@superset-ui/core/src/utils/featureFlags.ts:53:5 - error TS2717: Subsequent property declarations must have the same type.  Property 'featureFlags' must be of type 'FeatureFlagMap', but here has type 'FeatureFlagMap'.

53     featureFlags: FeatureFlagMap;
       ~~~~~~~~~~~~

  src/featureFlags.ts:54:5
    54     featureFlags: FeatureFlagMap;
           ~~~~~~~~~~~~
    'featureFlags' was also declared here.


Found 1 error.
```

This PR should fix it.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache-superset/superset-ui/pull/982
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/setup/setupApp.ts [hunk 1]
```diff
@@ -22,8 +22,16 @@ import { SupersetClient } from '@superset-ui/core';
...
+} from 'src/utils/getClientErrorObject';
+import setupErrorMessages from 'src/setup/setupErrorMessages';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+declare global {
+  interface Window {
+    $: any;
+    jQuery: any;
+  }
+}
```
Comment (etr2460) [inline superset-frontend/src/setup/setupApp.ts:34]: is this related to the feature flag change? i didn't see how

**Review by etr2460 — APPROVED** [review]
this is a bit inconvenient, but necessary because we moved featureflags into superset-ui and we don't want to change in two places everytime. stamping with one comment

**Review by villebro — APPROVED** [review]

File: superset-frontend/src/setup/setupApp.ts (hunk 1 above)
Comment (ktmud) [reply superset-frontend/src/setup/setupApp.ts:34]: It's not, but these globals used to be declared in the `featureFlags.ts` file. I moved them closer to where they are actually used.
